src/backend/base/langflow/components/vectorstores/OpenSearch.py

```diff
@@ -0,0 +1,292 @@
+
+    def search(self, query: Optional[str] = None) -> List[Dict[str, Any]]:
+        """
+        Search for similar documents in the vector store or retrieve all documents if no query is provided.
+        """
+        try:
+            vector_store = self.build_vector_store()
+            search_kwargs = {"k": self.number_of_results, "score_threshold": self.search_score_threshold}
+
+            if query:
+                search_type = self.search_type.lower()
+                if search_type == "similarity":
+                    results = vector_store.similarity_search_with_score(query, **search_kwargs)
+                    # Handle results with scores
+                    return [
+                        {"page_content": doc.page_content, "metadata": doc.metadata, "score": score}
+                        for doc, score in results
+                    ]
+                elif search_type == "mmr":
+                    results = vector_store.max_marginal_relevance_search(query, **search_kwargs)
+                    # Handle results without scores
+                    return [{"page_content": doc.page_content, "metadata": doc.metadata} for doc in results]
+                else:
+                    raise ValueError(f"Invalid search type: {self.search_type}")
+            else:
+                # Retrieve all documents when no query is provided
+                results = self.get_all_documents(vector_store, **search_kwargs)
+                return [
+                    {"page_content": doc.page_content, "metadata": doc.metadata, "score": score}
+                    for doc, score in results
+                ]
+        except Exception as e:
+            error_message = f"Error during search: {str(e)}"
+            logger.error(error_message)
+            logger.error(f"Traceback: {traceback.format_exc()}")
+            raise Exception(error_message)
+
+    def get_all_documents(self, vector_store: OpenSearchVectorSearch, **kwargs) -> List[Tuple[Document, float]]:
+        """
+        Retrieve all documents from the vector store.
+        """
+        try:
+            client = vector_store.client
+            index_name = vector_store.index_name
+
+            query = {"query": {"match_all": {}}, "size": kwargs.get("k", self.number_of_results)}
+
+            response = client.search(index=index_name, body=query)
+
+            results = []
+            for hit in response["hits"]["hits"]:
+                doc = Document(page_content=hit["_source"].get("text", ""), metadata=hit["_source"].get("metadata", {}))
+                score = hit["_score"]
+                results.append((doc, score))
+
+            return results
+        except Exception as e:
+            error_message = f"Error retrieving all documents: {str(e)}"
+            logger.error(error_message)
+            logger.error(f"Traceback: {traceback.format_exc()}")
+            raise Exception(error_message)
+
+    def search_documents(self) -> List[Data]:
```

Comment:
Curious if there was a specific use case that required the implementation of `search` here. We have a general (albeit, simplistic) implementation of search in the `LCVectorStore` parent that handles search.

Reply:
OpenSearch is capable of executing much more complicated kinds of hybrid searches than most databases.  We want to expose those features as much as possible.

Reply:
That makes sense. There may be changes here that are doing work I'm unaware of, so apologies if my comment doesn't apply -> it looks like there are no additional features added outside of those currently implemented by the default search in langchain, aside from changing the default behaviors (noted in other comment) in a few places. I would recommend either exposing the more complex features in this PR, and updating the implemented `search` appropriately`, _or_ removing this in this PR, then opening a new one that expands on its feature set.

Reply:
I just adapted the component so it could receive a hybrid search query as well.